delivery_carrier_label_seur/models/stock.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+# © 2015 FactorLibre - Ismael Calvo <ismael.calvo@factorlibre.com>
+# © 2017 PESOL - Angel Moya <angel.moya@pesol.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+try:
+    from seur.picking import Picking
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+try:
+    from unidecode import unidecode
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+
+class ShippingLabel(models.Model):
+    _inherit = 'shipping.label'
+
+    @api.model
+    def _get_file_type_selection(self):
+        """ To inherit to add file type """
+        res = super(ShippingLabel, self)._get_file_type_selection()
+        res.append(('txt', 'TXT'))
+        res = list(set(res))
+        return res
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    seur_service_code = fields.Selection(selection='_get_seur_services',
+                                         default=False)
+    seur_product_code = fields.Selection(selection='_get_seur_products',
+                                         default=False)
+
+    @api.model
+    def create(self, vals):
+        res = super(StockPicking, self).create(vals)
+        if 'carrier_id' in vals:
+            res.carrier_id_change()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(StockPicking, self).write(vals)
+        if 'carrier_id' in vals:
+            self.carrier_id_change()
```

Comment:
@jesusVMayor You cannot do that as self can be multi (see Travis error) and carrier_id_change is one